### PR TITLE
Add GD singleton and extensions

### PR DESCRIPTION
**Pull request by chippmann**
This adds the GD singleton and extensions @CedNaru implemented for the `godot-kotlin` project.

### CODE REVIEW

File: kt/godot-library/src/main/kotlin/godot/global/GD.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    override val rng = RandomNumberGenerator()
+
+    /** Asserts that the condition is true.
+    If the condition is false, an error is generated and the program is halted until you resume it.
+    Only executes in debug builds. Use it for debugging purposes, to make sure a statement is true during development. */
+    fun assert(condition: Boolean, message: String = "") = assert(condition) { message }
+
+    /** Returns whether instance is a valid object (e.g. has not been deleted from memory).*/
+    fun isInstanceValid(instance: Object): Boolean {
+        return instance.rawPtr != nullptr
```
Comment (chippmann) [inline kt/godot-library/src/main/kotlin/godot/global/GD.kt:30]: Here I'm unsure if this really is enough. If the pointer on the cpp side is gone this value is not updated as far as i can see.
Maybe some of you cpp devs have an idea how we can properly implement that?

**Review by piiertho — APPROVED** [review]

File: kt/godot-library/src/main/kotlin/godot/global/GD.kt (hunk 1 above)
Comment (CedNaru) [reply kt/godot-library/src/main/kotlin/godot/global/GD.kt:30]: For this one, you should probably use this function from the GarbageCollector:
https://github.com/utopia-rise/godot-jvm/blob/bf09d77505502fedacdd2ec0b3290d2adeb23509/kt/godot-runtime/src/main/kotlin/godot/core/GarbageCollector.kt#L188

Just checking if the ptr is null is not enough. Freeing the memory doesn't make the pointers to its location null.

Comment (piiertho) [reply kt/godot-library/src/main/kotlin/godot/global/GD.kt:30]: Good catch

**Review by piiertho — CHANGES_REQUESTED** [review]
Need fix of @CedNaru catch

**Review by CedNaru — CHANGES_REQUESTED** [review]
Not sure if he can actually use it because we need the id.

File: kt/godot-library/src/main/kotlin/godot/global/GD.kt (hunk 1 above)
Comment (chippmann) [reply kt/godot-library/src/main/kotlin/godot/global/GD.kt]: Added the suggested check. The `rawPointer` we store is the right one to use for the `checkInstance` check i assume?

**Review by CedNaru — APPROVED** [review]

**Review by piiertho — APPROVED** [review]

**Pull request merged**